book/src/actions.md

```diff
@@ -0,0 +1,221 @@
+With [`ButtonBuilder`](../docs/gtk4/struct.ButtonBuilder.html), we only have to add the corresponding method calls.
+
+<span class="filename">Filename: listings/actions/4/main.rs</span>
+
+```rust ,no_run,noplayground
+{{#rustdoc_include ../listings/actions/4/main.rs:button_builder}}
+```
+
+Actionable widgets are also easily accessible through the interface builder.
+As usual, we build up the window via a composite template.
+Within the template we can then set the "action-name" and "action-target" property.
+
+<span class="filename">Filename: listings/actions/5/window/window.ui</span>
+
+```xml
+{{#rustdoc_include ../listings/actions/5/window/window.ui}}
+```
+
+Connecting the actions and adding them to the window will be done in the method `Window::add_actions`.
+
+<span class="filename">Filename: listings/actions/5/window/mod.rs</span>
+
+```rust ,no_run,noplayground
+{{#rustdoc_include ../listings/actions/5/window/mod.rs:impl_window}}
+```
+
+Finally, `add_actions` will be called within `constructed`.
+
+<span class="filename">Filename: listings/actions/5/window/imp.rs</span>
+
+```rust ,no_run,noplayground
+{{#rustdoc_include ../listings/actions/5/window/imp.rs:object_impl}}
+```
+
+This app has the same behavior as the former one, but it leads the way to menus.
+
+
+## Menus
+
+If you want to create a [menu](https://developer.gnome.org/hig/patterns/controls/menus.html) you have to use actions and you will want to use the interface builder.
+Typically, a menu entry has an action fitting one of these three descriptions:
+- no parameter and no state, or
+- no parameter and boolean state, or
+- string parameter and string state.
+
+Let us modify our small app to demonstrate these cases.
+First we extend `add_actions`.
+For the action without parameter or state, we choose "win.quit" which we are already familiar with.
+
+
+<span class="filename">Filename: listings/actions/6/window/mod.rs</span>
+
+```rust ,no_run,noplayground
+{{#rustdoc_include ../listings/actions/6/window/mod.rs:action_quit}}
+```
+With the action "sensitive-button", we manipulate the "sensitive" property of `button`.
+We take care that we follow established conventions.
+Actions with no parameter and boolean state should behave like toggle actions.
+That means, that the caller can expect that after they activated the action, the boolean state will toggle.
+
+<span class="filename">Filename: listings/actions/6/window/mod.rs</span>
+
+```rust ,no_run,noplayground
+{{#rustdoc_include ../listings/actions/6/window/mod.rs:action_sensitive_button}}
+```
+
+Finally, we add "win.orientation", an action with string parameter and string state.
+This action can be used to change the orientation of `gtk_box`.
+Here the convention suggests that the state of stateful actions with a parameter, should follow the given parameter.
+The logic of the code does not rely on a state but with it, the menu will be able to display the current state.
```

Comment:
The point is that we actually don't need a stateful action, because we just set the orientation to the parameter.
The only benefit is that then the menu knows what the current state is and can display it.
Do you have an idea how to make this point clearer or should I just drop it?

Reply:
Something like "We don't need the action state to implement orientation switching, however it is useful for making the menu display the current orientation."?